updater/lib/dependabot/updater/operations/refresh_security_update_pull_request.rb

```diff
@@ -0,0 +1,233 @@
+# frozen_string_literal: true
+
+# This class implements our strategy for 'refreshing' an existing Pull Request
+# that updates an insecure dependency.
+#
+# TODO: copyedit
+#
+# It will determine if the existing diff is still relevant, in which case it
+# functions similar to a "rebase", but in the case where the project folder's
+# dependencies have changed or a newer version is available, it will supersede
+# the existing pull request with a new one for clarity.
+module Dependabot
+  class Updater
+    module Operations
+      class RefreshSecurityUpdatePullRequest
+        include SecurityUpdateHelpers
+
+        def self.applies_to?(job:)
+          return false unless job.security_updates_only?
+          # If we haven't been given metadata about the dependencies present
+          # in the pull request, this strategy cannot act.
+          return false if job.dependencies&.none?
+
+          job.updating_a_pull_request?
+        end
+
+        def self.tag_name
+          :update_security_pr
+        end
+
+        def initialize(service:, job:, dependency_snapshot:, error_handler:)
+          @service = service
+          @job = job
+          @dependency_snapshot = dependency_snapshot
+          @error_handler = error_handler
+        end
+
+        def perform
+          dependency = dependencies.last
```

Comment:
Why not `.first` which is the lead dependency? I think we refer to it in other places, might make sense to be consistent with error messaging. Or was this how it has always been?

Reply:
> Or was this how it has always been?

Yes, for some reason we have always used first for `lead_dependency` but last here. There are three places we select 1 of N dependencies, 2 are `last` and 1 is `first` and I'm not 100% sure why.

I will scan back through the code and add some TODOs on this though as I think I earmarked it in the prototype way back and didn't leave a note in the actual implementation.

Reply:
No harm in leaving it then, don't want to introduce a new bug somehow!